### DISCUSSION TITLE
this isn't at 66% win rate

### DISCUSSION THREAD

**Discussion by aaaclones** [kyleskom/NBA-Machine-Learning-Sports-Betting#153] [General]
how is this model 66% win rate ?

**Comment by kyleskom** [comment]
I don't understand the question

**Comment by aaaclones** [comment]
Hey  Kyle ,

For the Xg boost model prediction says it predicts at a 67% ml win rate ?
52.5 on under/over ?

**Reply by kyleskom** [reply]
We've been training a bunch more models lately and things have changed. I have to go back and revisit how I got my original results.

**Comment by aaaclones** [comment]
Do you update the model daily ?

**Reply by JP23000** [reply]
Yes you should.  It pulls data and if you don't update daily you wont have yesterday's stats.

**Comment by aaaclones** [comment]
Kyle can you tell me what prediction is more accurate ? how can the delta
between the two be so despair ?

**Reply by JP23000** [reply]
Hey aaclones,

The difference between Money Line and OU results?  Is that what you are trying task?

**Reply by JP23000** [reply]
If so... It has to do with the implied probability of events happening.  Vegas tends to get ML picks correct in the NBA about 67% of the time, with odds ranging from -105 to -1400 for probability.  It is easier to get a higher accuracy when the assumed winner is more obvious.

The OU comes with a -110 odds rate because it is the mean of what they modeled.  The hit rate for this is 50/50, with a 52.5% rate needed to cover Vegas.  Anything above that is killer.  My OU model is 61% over the last 5 days.  Just blowing away Vegas.

**Reply by kyleskom** [reply]
Absolutely love to hear it!

**Reply by skuan** [reply]
How are you applying the EV number that the model spits out to the OU bet? I get for ML you just bet on the team with positive EV, but wasn't sure how that translates to a point spread.

**Comment by JP23000** [comment]
So since 3/15/2023 I am 66.66% on ML and 61.9% on OU.

**Reply by JP23000** [reply]
The updated daily model has been lights out!!!!

**Comment by aaaclones** [comment]
i want to take a moment to thank you for you work ? i want you to be
compensated how do i go about doing so ?

**Comment by aaaclones** [comment]
The totals more so ... for example sometimes the Xg boost model has a
absurd delta from the neural network , the neural network seems to be a lot
more on the conservative side . I am trying to figure out what seems to be
more accurate as i been monitoring it the last few days

**Reply by JP23000** [reply]
Gotcha, yeah the XG Boost is the more accurate one to go with.

**Comment by aaaclones** [comment]
private message me , if possible or send me crypto wallet or anyway  I
compensate you for this amazing project

**Reply by kyleskom** [reply]
You really don't have to.

**Comment by aaaclones** [comment]
no it states on the xg boost model ratings that it's 66% and 52.5 on the
o/u , but Kyle already cleared it up

**Comment by aaaclones** [comment]
something seems off with the model the last few days the
statistical significance is very alarming , the totals are just very off
...

**Comment by aaaclones** [comment]
for example it gave there are 3 games in the 60 percentile it should win
3-1 , thats not the case and the ev is not even close

**Comment by aaaclones** [comment]
This model isn't giving you Ev for o/u it just gives you a % on what it
thinks it going to happen in totals but it hasn't been as accurate as it
once was , and to further your question this has nothing to do with spreads
this purely gives you Ev Values if you were to make this bet over a course
of time you will be positive E/v but the model needs some tweaking since
the last few days as you can see the model has been getting blown out the
waters

**Reply by skuan** [reply]
I've been noticing some ridiculously high EVs lately but I think that might be due to injuries or teams shutting down their star players and going into "tank" mode as the season starts to wind down.  There are at least 3 teams where I consistently see EVs in the 80+ range that have benched most of their starters.

**Reply by nealmick** [reply]
I agree, noticed a lot of players on injury report.  Beginning and ending of season is unreliable.

**Reply by skuan** [reply]
I'm kind of okay with this since the requirement of some human judgment ensures that this machine learning doesn't completely break the NBA betting ecosystem.

**Reply by nealmick** [reply]
I have read online bookmakers like Draftkings and Fanduel will limit or ban if you profit in the long run.
https://www.bettingusa.com/sportsbooks-ban-smart-customers/
Mostly I think people limited for arbitrage, but thats easily detectable with specific bet amounts like 3284.51$

I think we look like normal bettors, that just win more often than not, it would take a significant amount won before ban from what ive read... the unlimited money printer thing is not true even with a profitable model.  Its similar to card counting, not illegal, but once they detect you winning too much they will limit you or ask you to take your business else where... Its not like you only get one chance tho, there are hundreds of casinos and bookmakers.

**Reply by kyleskom** [reply]
There's going to be a lot of variance in shorter periods of time. As for EV yes, players sitting out has a huge impact on this. It's always best to look at lineups for other things and make adjustments on your own. This is just a single tool.

**Reply by JP23000** [reply]
I hit 60% playing every O/U each day from 3/1 on when I discovered this repository.  Made 10k betting $100 a bet.  Let's get the ability to scrape the player lineup and I can supply the last 7 years of NBA player data that I got from Big Data Ball.

**Reply by JP23000** [reply]
As for bettor bans.... I will figure out where the article is, but they are not really concerned with people who use systems to win, at least to date.  This may change with the proliferation of AI models, but I doubt it.  IN some instances yes they ban you or restrict your amount to bet... but this is generally people who are winning really long shot odds bets that shouldn't have hit... and was suppose to be the money maker for the casino.  And those they find that are gaming the system by doing something illegal like using other people to place bets for them (Like this dude who had a 17 leg parlay about to hit in the MLB playoffs, but he lived in a state that it was illegal to place a bet and there was no record of him being in the state where the bet was placed.)  Or people finding arbitrage situations.

Generally speaking with O/U the sportsbooks main goal is to maintain a balanced book.  They set the line and then adjust odds and move the line according to where the bets are coming in at so that they can offset them.  That's why the odds are -110 instead of -100 or +100.  When vegas maintains a balanced book they make 5% no matter the outcome.

Really this is those on the right side of the bet vs those on the wrong side of the bet.  At least this is what I learned when I interviewed for their US Director for Improvements that they had open a while back.

**Reply by nealmick** [reply]
I disagree, No one that remains profitable for a long enough period is aloud to play.  The people that win big parleys will never get limited, why would you limit someone consistently giving money on losing bets?? they get lucky once like the lottery, and then lose it right back.  The people that beat closing lines consistently get limited, "sharps" if you wana call it that, and a 10 leg parley isn't very sharp.  https://github.com/kyleskom/NBA-Machine-Learning-Sports-Betting/discussions/161

The only odds maker that doesn't really limit and is widely trusted is pinnacle sports.

**Comment by aaaclones** [comment]
You don’t have any scraping of updating  lineups since players in/out skews
the data tremendously, and the whole key of proper data is to some extent
minimize variance as much as possible , players in or out actually
increases variance

**Reply by kyleskom** [reply]
Sure does. Adding player data is something im actively thinking about and wanting to work on. Just need the time.

**Reply by jbolter** [reply]
Are you looking for help in adding player data? I'm new in this arena but this project is exactly what I was thinking about creating (with player data).  i have a CS background but am green in the ML arena.

**Reply by kyleskom** [reply]
@jbolter Yes I am looking for help in incorporating player data.

**Reply by nealmick** [reply]
That would be amazing, i have something started here https://github.com/nealmick/Merge-Data
Basically just merges data from my project and writes top 7 players by play time to team data csv. ill link my project below aswell.
https://github.com/nealmick/bb

**Comment by aaaclones** [comment]
is there a statistical significance in the case the model starts to be
really off and correction needs to be made ? Is there a threshold number ?
Youve done amazing work I am just wondering if and when it gets majorly
stressed test will the corrections be made ??

**Reply by kyleskom** [reply]
Working on some small things now dance cleaning things up. Will hopefully get it it soon.

**Reply by BG2011** [reply]
This project on his own has Made me to take the decission to start venturing into AI. Thanks man ! I have Made fake bets / combi’s with Some nba knowledge and this tool and have for this matchday a completion rate of 80!!!% even a 45odd combi was correct. And this is without the new model. This is a gem 👌

**Comment by aaaclones** [comment]
How will you guys update daily for this ? Are there any reliable sources
for api keys ?

**Reply by nealmick** [reply]
balldontlie.io

**Comment by aaaclones** [comment]
Last time I checked balldontlie didn’t

**Reply by kyleskom** [reply]
Pulling data from nba directly is fairly easy

**Reply by nealmick** [reply]
this is true!

**Comment by aaaclones** [comment]
And how do you intend to fetch the data for each player daily ??

**Reply by nealmick** [reply]
https://nbadata.cloud/predict/player-details/237

**Comment by nealmick** [comment]
It might be easier to input the last game for each team.  Would probably offer some performance.  Just basically forming last game of each team same as usual would  3x the input data features, and offer alot of momentum short-term type data, kinda like streaks.  Could make it trailing 2 or 3 games as-well.  Along the same lines, the last game against the opponent team could be added for input as-well.  These changes would make the first game of the season break, unless you track last game over multiple seasons, kinda difficult, i just removed first game's for my project.

**Reply by nealmick** [reply]
Also another easy change would be to add categorical inputs for each team id.  This would allow the model to tell one team from another.

**Reply by nealmick** [reply]
@kyleskom what do you think of these potential changes ^^

**Reply by kyleskom** [reply]
I think that would be beneficial for the top part. I don't like to know the team and we just predict based on teams stats. I think that could skew the results. Ill think on that as well.

**Reply by nealmick** [reply]
I agree, probably would help account for injuries too if last game performance was bad.  The main negative thing is it drives the API request up quickly.

**Comment by aaaclones** [comment]
i've tried a bunch of inputs it's so far off ... which categorical inputs ?
and how do you update each player individually on a regular basis ? that
player might not even be playing that day .. i cant find one api key that
allows me to automate

**Reply by nealmick** [reply]
team id categorical input.  Please take some time to review API endpoints.
https://rapidapi.com/tank01/api/tank01-fantasy-stats/

**Comment by aaaclones** [comment]
Neal , do you plan on doing it the exact same way but use player stats ?
Ive tried 4 libraries they are underperforming , i am trying to customize
some advance stats currently but again the problem is the api keys , the
rapid api end point doesn't provide the advanced stats for all the players
, nbastats does but can't automate with them

**Reply by nealmick** [reply]
I aggregate and correlate data across multiple api's, both historical and live.

**Reply by deperto** [reply]
> Neal , do you plan on doing it the exact same way but use player stats ? Ive tried 4 libraries they are underperforming , i am trying to customize some advance stats currently but again the problem is the api keys , the rapid api end point doesn't provide the advanced stats for all the players , nbastats does but can't automate with them
> […](#)

What about this: https://github.com/swar/nba_api

**Reply by nealmick** [reply]
From my experience the stats.nba.com endpoint ip bans if you make too many requests. I have also found request time to very too much, sometimes taking over 30 seconds to reply making page loads of new games too long.

**Comment by aaaclones** [comment]
this doesn't include advanced stats and i believe it's not uptodate

**Reply by nealmick** [reply]
Includes many advanced stats, pulled directly from nba.com

**Reply by kyleskom** [reply]
What needs to be updated exactly? The predictions are used with the most updated team stats every single time you run.

**Comment by aaaclones** [comment]
How about for player stats ? In the event they aren’t in the lineup it
skews the whole entire data batch from the team we want to minimize
variance as much as possible , so I feel it’s important to do daily player
stats to match up with the team stats , I’ve been trying but fetching the
player stats is hard to automate

**Reply by kyleskom** [reply]
This doesn't include player data and it skewed by injuries yes. I plan on working on that for next season.

**Comment by aaaclones** [comment]
I can pitch in I can show you what I’ve done so far the only problem I get
is automating for now I manually punch in the player data and it’s so
tedious , the last 3 days using what you have and including what I did I
successfully predicting 4 games the exact score but I am not sure if it
pure luck or not since my player predictions weren’t accurate

**Comment by aaaclones** [comment]
Team stats will only remain true if the standard deviation is constant ,
but as you see in nba its not and when a key player is out the effect it
has is linear , there needs to be a solution of some sort for these
scenarios.. I’ve explored a lot ways to try and solve this but the hard
part is automated the day to day data to ensure you know whose in or whose
out and updating the data since all these api don’t work even the expensive
ones , unless you manually just punch it in it’s been a tall order finding
a way

**Reply by nealmick** [reply]
Most likely multifactorial, and only linear on a hyperplane described by the coefficients slope and intercept.

**Reply by kyleskom** [reply]
I'll work on this and have it fully automated for next season. I have the idea of how to do this.

**Comment by aaaclones** [comment]
The top 3 players on the team will cause this

**Comment by aaaclones** [comment]
Incredible , cant wait !!!
